File: ApiBaseCore/ApiBaseCore.Application/DTOs/PerfilUsuarioDto.cs
using System.Text.Json.Serialization;

namespace ApiBaseCore.Application.DTOs;

/// <summary>
/// Representa los datos del perfil de un usuario que son seguros para ser devueltos por la API.
/// </summary>
public class PerfilUsuarioDto
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public string Phone { get; set; }
    [JsonPropertyName("tax_id")]
    public string? TaxId { get; set; }
    [JsonPropertyName("created_at")]
    public string CreatedAt { get; set; }
    public string? AvatarUrl { get; set; }
    public string Rol { get; set; }
    public List<AddressDto> Addresses { get; set; } = new();
}

File: ApiBaseCoreJava/src/main/java/com/polancou/apibasecore/application/dtos/PerfilUsuarioDto.java
package com.polancou.apibasecore.application.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Data
public class PerfilUsuarioDto {
    private UUID id;
    private String name;
    private String email;
    private String phone;
    
    @JsonProperty("tax_id")
    private String taxId;
    
    @JsonProperty("created_at")
    private String createdAt;
    
    private String avatarUrl;
    private String rol;
    
    private List<AddressDto> addresses = new ArrayList<>();
}
